draft-boucadair-add-deployment-considerations.txt

```diff
@@ -588,7 +588,9 @@ Internet-Draft    Discovery of Encrypted DNS Resolvers    September 2022
 4.  Hosting Encrypted DNS Forwarder in Local Networks
 
    This section discusses some deployment considerations to host an
-   encrypted DNS forwarder within a local network.
+   encrypted DNS forwarder within a local network.  DDR does not apply
```

Comment:
DDR is not applicable for private IPv4 addresses used but can be used with public IPv6 addresses used by the CPE. We should probably discuss that if the CPE can be upgraded to host a encrypted DNS forwarder it should prefer to get domain name validated certificate instead of requesting the issuance of a IP address attested certificate. As of writing of this specification, CAs do not support ACME IP Identifier Validation Extension attesting to IP addresses (https://www.rfc-editor.org/rfc/rfc8738.html).

Reply:
I'm not certain we should go so far as to recommend one approach over another. We should leave this to implementors since the argument you're presenting may change over time (as more ACME validators support RFC8738) and doesn't apply to non-ACME scenarios, which is common for larger organizations. The large DNS resolvers for example all have non-ACME certs that already claim their IP addresses. Large enterprises may do the same for themselves.

Perhaps we need a separate PR to do a consideration point about the trade-offs of these approaches?

Reply:
Another trade-off to add to that list:  The ISP can renumber the subscriber's IPv6 prefix on a whim, breaking IPv6-address-based DDR validation until the CPE fetches itself a new certificate for that new address.

Reply:
The other trade-off is CPE has to run a HTTP server exposing it to external attacks till the IP address validation procedure is complete. The other problem is ISP assigned IPv6 prefix is not a stable identifier unlike a FQDN. DNS client implementations will not be able to take advantage of https://datatracker.ietf.org/doc/html/draft-wing-opsawg-authenticating-network-00.

Reply:
With Dan's ongoing stab at this list offline, can we consider this comment closed for this PR (expecting his changes to come via another PR)?

Reply:
No, DDR is applicable to this section as well. You may want to remove the following text: DDR does not apply to any of these scenarios for the same reason it does not apply to any of the proxied DNS scenarios.